wallet/txauthor/author.go

```diff
@@ -48,7 +48,10 @@ type AuthoredTx struct {
 
 // ChangeSource provides P2PKH change output scripts and versions for
 // transaction creation.
-type ChangeSource func() ([]byte, uint16, error)
+type ChangeSource interface {
+	Source() ([]byte, uint16, error)
```

Comment:
Rename this to Script, and perhaps name the return values so it's more clear the middle one is the version.